source/classification1.md

```diff
@@ -1727,27 +1626,16 @@ Despite the simplicity of the problem, solving it in a statistically sound manne
 fairly nuanced, and a careful treatment would require a lot more detail and mathematics than we will cover in this textbook.
 For the present purposes, it will suffice to rebalance the data by *oversampling* the rare class. 
 In other words, we will replicate rare observations multiple times in our data set to give them more
-voting power in the $K$-nearest neighbor algorithm. In order to do this, we will need an oversampling
-step with the `resample` function from the `sklearn` Python package.
-We show below how to do this, and also
-use the `.groupby()` and `.count()` methods to see that our classes are now balanced:
-
-```{code-cell} ipython3
-:tags: [remove-cell]
-
-# Despite the simplicity of the problem, solving it in a statistically sound manner is actually
-# fairly nuanced, and a careful treatment would require a lot more detail and mathematics than we will cover in this textbook.
-# For the present purposes, it will suffice to rebalance the data by *oversampling* the rare class. \index{oversampling}
-# In other words, we will replicate rare observations multiple times in our data set to give them more
-# voting power in the $K$-nearest neighbor algorithm. In order to do this, we will add an oversampling
-# step to the earlier `uc_recipe` recipe with the `step_upsample` function from the `themis` R package. \index{recipe!step\_upsample}
-# We show below how to do this, and also
-# use the `group_by` and `summarize` functions to see that our classes are now balanced:
-```
-
-```{code-cell} ipython3
-rare_cancer['Class'].value_counts()
-```
+voting power in the $K$-nearest neighbor algorithm. In order to do this, we will 
+first separate the classes out into their own data frames by filtering.
+Then, we will
+use the [`resample`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.resample.html) function 
+from the `sklearn` package to increase the number of `Malignant` observations to be the same as the number 
+of `Benign` observations. We set the `n_samples` argument to be the number of `Malignant` observations we want. 
+We also set the `random_state` to be some integer
+so that our results are reproducible; if we do not set this argument, we will get a different upsampling each time
+we run the code. Finally, we use the `value_counts` method
+ to see that our classes are now balanced.
```

Comment:
Here we introduce random state, then in the next chapter we talk about seeds instead.

Reply:
good point. will edit that to remove mention and hide a call to `np.random.seed` internally.